hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -403,16 +416,32 @@ public TransactionContext startTransaction(RaftClientRequest request)
           .setStateMachineData(write.getData())
           .setLogData(commitContainerCommandProto.toByteString())
           .build();
-    } else {
-      return TransactionContext.newBuilder()
+    } else if (proto.getCmdType() == Type.FinalizeBlock) {
+      containerController.addFinalizedBlock(proto.getContainerID(),
```

Comment:
We cannot add the finalize block in startTransaction, otherwise there is chance that valid write chunk or put block in applyTransaction will be failed because of this.

Reply:
We are rejecting request only in `startTransaction` and not in `applyTransaction`. So once the request has passed in startTransaction I think it should succeed in `applyTransaction.` In which case this can happen?

Reply:
I see.  Finalized block is not checked in applyTransaction.